packages/blade/src/components/Text/Decisions/Decisions.md

```diff
@@ -0,0 +1,87 @@
+The text component is categorised into multiple parts
+
+- BaseText - Internal Component, not published to consumers
+- Title - Public Component
+- Heading - Public Component
+- Text - Public Component
+- Code - Public Component
+
+Here's how the hierarchy will looks like
+![Component Breakdown](./component-breakdown.png)
+
+The APIs for all the above component will look something like below
+
+- `BaseText`
+
+  This component is just meant to be used for internal purposes for Blade
+  Here's how the API will look like
+
+  ```jsx
+  <BaseText
+    color="surface.text | feedback.text | button.text"
+    fontFamily="fontfamily.name"
+    fontSize="fontSize.name"
+    fontWeight="fontWeight.name"
+    lineWeight="lineWeight"
+    letterSpacing="letterSpacing"
+    formatting="bold/italic/strike"
+    as="h1-h6/p/span"
+  />
+  ```
+
+- `Title`
+
+  This component is meant to be used for Titles(bigger banner style kind of text) mostly on landing pages or page level headings.
+
+  Title will internally use `BaseText` to render. We'll use native html heading tags to render title and native text element for react-native to render the Title.
+
+  ```jsx
+  <Title variant="small | medium | large" type="normal | subtle | subdued | muted | placeholder">
+    Some Title
+  </Title>
+  ```
+
+- `Heading`
```

Comment:
we basically want to free people from deciding H1-H6 as we are using a design language to streamline this vocabulary. so H1-H6 is understandable for web folks but not by designers and react native developers. So since we'll also be publishing for react-native there's no such thing as H1-H6 for react native it's all plain text. Regarding, your question about how we'll identify which H1 to render on the web we have variants if you see them in the API. We'll have a variant to equivalent Heading tag internally and take that decision internally while rendering instead of consumers taking that call. So as a consumer(dev/designer) I'll just say I need small heading, medium heading or large heading.

Reply:
aha that sounds better. We can see how it acts out in actual applications and make changes later if accessibility is getting imacted.